apps/prairielearn/elements/pl-file-upload/pl-file-upload.js

```diff
@@ -127,18 +192,20 @@
 
         var commaSplitIdx = dataUrl.indexOf(',');
         if (commaSplitIdx === -1) {
-          this.addWarningMessage('<strong>' + name + '</strong>' + ' is empty, ignoring file.');
+          this.addWarningMessage(
+            `<strong>${escapeFileName(name)}</strong> is empty, ignoring file.`,
+          );
           return;
         }
```

Comment:
In this case, the file name would have been removed from the list of required regexes (lines 161-168), but not added to the list, so the state of the data would be inconsistent.

Reply:
See below.